### PR TITLE
add core.request.get_scheme/host/port/http_version

### PR DESCRIPTION
**Pull request by taotao2014**
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

**Comment by moonming** [comment]
@taotao2014 please fix conflicting files: apisix/core/request.lua

File: apisix/core/request.lua [hunk 1]
```diff
@@ -94,7 +94,34 @@ function _M.get_remote_client_port(ctx)
...
+end
+
+function _M.get_port(ctx)
+    if not ctx then
+        ctx = ngx.ctx.api_ctx
+    end
+    return tonumber(ctx.var.server_port)
+end
+
+function _M.get_http_version()
```
Comment (membphis) [inline apisix/core/request.lua:120]: How about `_M.get_http_version = ngx.req.http_version` ?

Comment (taotao2014) [reply apisix/core/request.lua:120]: Method is better

Comment (membphis) [reply apisix/core/request.lua:120]: I think `_M.get_http_version = ngx.req.http_version` is better, simpler and faster.

**Comment by taotao2014** [comment]
> @taotao2014 please fix conflicting files: apisix/core/request.lua

Only those with write access to this repository can merge pull requests.

**Comment by taotao2014** [comment]
> @taotao2014 please fix conflicting files: apisix/core/request.lua

Only those with write access to this repository can merge pull requests.

File: apisix/core/request.lua (hunk 1 above)
Comment (taotao2014) [reply apisix/core/request.lua:120]: Keep the style consistent,user acceptance is easier.

File: apisix/core/request.lua [hunk 2]
```diff
@@ -94,7 +94,34 @@ function _M.get_remote_client_port(ctx)
...
+
```
Comment (membphis) [inline apisix/core/request.lua:121]: remove this blank line

File: apisix/core/request.lua [hunk 3]
```diff
@@ -94,7 +94,34 @@ function _M.get_remote_client_port(ctx)
...
+    return ngx.req.http_version()
+
```
Comment (membphis) [inline apisix/core/request.lua:123]: ditto

File: apisix/core/request.lua (hunk 1 above)
Comment (membphis) [reply apisix/core/request.lua:120]: > Keep the style consistent,user acceptance is easier.

We always need APISIX core to have good performance, that is very important.

File: apisix/core/request.lua (hunk 3 above)
Comment (taotao2014) [reply apisix/core/request.lua]: removed

**Comment by membphis** [comment]
https://github.com/apache/apisix/pull/1953/files#diff-118ffaea87717e572415a97fb2a95e57R119-R146

It seems you are doing something wrong when using git. I think you can submit your code by new PR.

**Pull request closed without merging**